mmpose/core/evaluation/top_down_eval.py

```diff
@@ -250,6 +250,83 @@ def _taylor(heatmap, coord):
     return coord
 
 
+def post_dark(coords, batch_heatmaps, share_heatmap=True):
+    """DARK post-pocessing.
+
+    Note:
+        batch_size: N
+        num_keypoints: K
+
+    Args:
+        coords (np.ndarray[N, K, 2]): Coords in shape of batchsize*num_kps*2.
+        batch_heatmaps (np.ndarray[N, K, H, W]): batch_heatmaps
+        share_heatmap (bool) : True for bottom_up paradigm where all persons
+                    share the same heatmap while False for top_down paradigm
+                    where each person has its own heatmaps.
+
+    Returns:
+        res (np.ndarray): Refined coords in shape of batchsize*num_kps*2.
+    """
+    if not isinstance(batch_heatmaps, np.ndarray):
+        batch_heatmaps = batch_heatmaps.cpu().numpy()
+    batch, num_kps, h, w = batch_heatmaps.shape
+    persons = coords.shape[0]
+    if share_heatmap:
+        assert batch == 1
+    else:
+        assert batch == persons
+    for i in range(batch):
+        for j in range(num_kps):
+            mapij = batch_heatmaps[0, j, :, :]
+            mapij = cv2.GaussianBlur(mapij, (3, 3), 0)
+            batch_heatmaps[0, j, :, :] = mapij
+    batch_heatmaps = np.clip(batch_heatmaps, 0.001, 50)
+    batch_heatmaps = np.log(batch_heatmaps)
+    batch_heatmaps = np.transpose(batch_heatmaps, (2, 3, 0, 1)).reshape(
+        (h, w, batch * num_kps))
+    batch_heatmaps_pad = cv2.copyMakeBorder(
+        batch_heatmaps, 1, 1, 1, 1, borderType=cv2.BORDER_REFLECT)
+    batch_heatmaps_pad = np.transpose(
+        batch_heatmaps_pad.reshape((h + 2, w + 2, batch, num_kps)),
+        (2, 3, 0, 1))
+    index = coords[:, :, 0] + 1 + (coords[:, :, 1] + 1) * (w + 2)
+
+    index = index.reshape((persons, -1))
```

Comment:
needless brackets

Reply:
btw why do you need this reshape?

Reply:
meet the requirement of shape for the concatenate

Reply:
I guess the index already has this shape

Reply:
yes, it has, and the 'reshape' operation is unnecessary